app/steps/residential_address.rb

```diff
@@ -10,18 +10,35 @@ class ResidentialAddress < Step
     :zip,
   )
 
-  validates :street_address,
-    presence: { message: "Make sure to provide a street address" }
+  with_options if: :stable_housing? do |address|
+    address.validates :street_address,
+      presence: {
+        message: "Make sure to provide a street address",
+        allow_blank: false,
```

Comment:
It defaults to `true` if you leave this out. You need to set this to false if you want to reject empty strings. Noticed this when I was going back and forth between this page, and the next page. Hard to explain what I saw - but consider this, the first visit to this page before submitting the values will be `nil`. Subsequent visits will be `""` so there's a clear difference in behavior unless you tell this here to NOT allow blanks.